lib/browser/api/auto-updater/squirrel-update-win.ts

```diff
@@ -15,89 +15,67 @@ const isSameArgs = (args: string[]) => args.length === spawnedArgs.length && arg
 
 // Spawn a command and invoke the callback when it completes with an error
 // and the output from standard out.
-const spawnUpdate = function (args: string[], detached: boolean, callback: Function) {
-  let error: Error, errorEmitted: boolean, stderr: string, stdout: string;
-
-  try {
+const spawnUpdate = async function (args: string[], detached: boolean): Promise<string> {
+  return new Promise((resolve, reject) => {
     // Ensure we don't spawn multiple squirrel processes
     // Process spawned, same args:        Attach events to already running process
     // Process spawned, different args:   Return with error
     // No process spawned:                Spawn new process
     if (spawnedProcess && !isSameArgs(args)) {
-      return callback(`AutoUpdater process with arguments ${args} is already running`);
+      throw new Error(`AutoUpdater process with arguments ${args} is already running`);
```

Comment:
I think we might instead want to `reject` here?

Reply:
this will reject correctly